src/twisted/internet/defer.py

```diff
@@ -159,6 +165,65 @@ def maybeDeferred(f, *args, **kw):
         return succeed(result)
 
 
+def unbox(deferred, raises=True, passthrough=True):
```

Comment:
I'm not sure about the ``passthrough`` argument here.

Essentially, the problem boils down to if you don't have ``passthrough=False`` then when the ``Deferred`` gets garbage collected if it was in a failure state then it prints a spurious traceback to stdout. This sees to be a problem that other utilities like ``gatherResults`` has had and they have a ``consumeErrors`` parameter which defaults to ``False`` but is documented that you should almost always use ``consumeErrors=True``.

So I guess the question is, what behavior with results makes sense?

* Always pass through results, success or failure and if someone wants something different then they need to add their own callback that throws away the result.
* Always pass through successful results but consume any failure results under the assumption that if you've unboxed the ``Deferred`` then you've handled the failure (and if not, you can add your own callback that re-inserts it).
* Never pass through any result and "break" the callback chain if someone continues to re-use the ``Deferred`` instance after you've unboxed it. If you want that to continue working then you'll need to add your own callback that re-inserts the result.
* Add a ``passthrough`` parameter that controls in both the success and the failure case whether or not it should be passed through or not.
    * If this is the answer, should we pass through results (success and failure) by default or not?
* Add a ``consumeErrors`` parameter that controls whether or not ``unbox()`` will consume errors.
  * Should this default to ``False`` as the other utility functions do (though I think they do for backwards compatibility) or should this default to ``True`` to more likely do the right thing?
  * Should we always return success results then or should we always consume them? Does it make sense to add a ``consumeSuccesses`` parameter for success results? If so what should it default to?